numpy/distutils/clang_cl.py

```diff
@@ -0,0 +1,167 @@
+    """
+
+    compiler_type = "clang-cl"
+    compiler_cxx = "clang-cl"
+
+    def __init__(self, verbose=0, dry_run=0, force=0):
+        super().__init__(verbose, dry_run, force)
+
+    def initialize(self):
+        # The 'lib' and 'include' variables may be overwritten
+        # by MSVCCompiler.initialize, so save them for later merge.
+        environ_lib = os.getenv("lib", "")
+        environ_include = os.getenv("include", "")
+        MSVCCompiler.initialize(self)
+
+        self.cc = _find_exe("clang-cl.exe")
+        self.linker = _find_exe("lld-link.exe")
+        if self.cc is None:
+            raise FileNotFoundError(
+                "Unable to locate clang-cl.exe. It should be on the path"
+            )
+        for opt in ("/GL", "/GL-", "/Ox"):
+            if opt in self.compile_options:
+                self.compile_options.remove(opt)
+        ltcg = "/LTCG"
+        ldflags = self._ldflags
+        for key in ldflags:
+            if ltcg in ldflags[key]:
+                ldflags[key].remove(ltcg)
+        self.compile_options.extend(
+            [
+                "/O2",
+                "/GS-",
+                "-Wno-visibility",
+                "-Wno-logical-op-parentheses",
+                "-Wno-microsoft-include",
+                "-Wno-shift-op-parentheses",
+            ]
+        )
+
+        # Merge current and previous values of 'lib' and 'include'
+        os.environ["lib"] = _merge(environ_lib, os.environ["lib"])
+        os.environ["include"] = _merge(environ_include, os.environ["include"])
+        # Get version information about clang which is needed to find
+        # the correct include and lib directories
+        clang_base = os.path.split(self.cc)[0]
+        clang_version_cmd = [self.cc, "--version"]
+        # Get version or let the user know if there was an issue
+        try:
+            clang_full_ver = subprocess.check_output(
+                clang_version_cmd
+            ).decode()
+        except WindowsError:
+            raise WindowsError(
+                "clang-cl.exe could not be found. It must be on your path."
+            )
+        clang_version = ""
+        if "version" in clang_full_ver:
+            loc = clang_full_ver.find("version")
+            clang_ver_string = clang_full_ver[loc + 7:]
+            clang_version = clang_ver_string.split("\n")[0].strip()
+        clang_command_and_return = (
+            f"Running the command \n\n{' '.join(clang_version_cmd)}\n\n "
+            f"which returned \n\n{clang_full_ver}"
+        )
+
+        if not clang_version:
+            raise RuntimeError(
+                f"The clang version could not be detected from the version "
+                f"string. {clang_command_and_return}"
+            )
+
+        # Forbid compiling 32 on 64-bit LLVM and vice versa
+        target = "i686" if platform_bits == 32 else "x86_64"
+        if target not in clang_full_ver:
+            raise RuntimeError(
```

Comment:
This makes sense. Would be good to add to the comment that `platform_bits` represents the Python interpreter, not the OS.

Reply:
Comment expanded.